UI/window-basic-main-outputs.cpp

```diff
@@ -761,90 +765,109 @@ bool SimpleOutput::SetupStreaming(obs_service_t *service)
 		auth->OnStreamConfig();
 
 	/* --------------------- */
-
-	const char *type = obs_service_get_output_type(service);
-	if (!type) {
-		type = "rtmp_output";
-		const char *url = obs_service_get_url(service);
-		if (url != NULL &&
-		    strncmp(url, FTL_PROTOCOL, strlen(FTL_PROTOCOL)) == 0) {
-			type = "ftl_output";
-		} else if (url != NULL && strncmp(url, RTMP_PROTOCOL,
-						  strlen(RTMP_PROTOCOL)) != 0) {
-			type = "ffmpeg_mpegts_muxer";
+	bool resetOutputs = true;
+	std::string streamOutputType;
+
+	for (auto &service : services) {
+		const char *type = obs_service_get_output_type(service);
+		if (!type) {
+			type = "rtmp_output";
+			const char *url = obs_service_get_url(service);
+			if (url != NULL && strncmp(url, FTL_PROTOCOL,
+						   strlen(FTL_PROTOCOL)) == 0) {
+				type = "ftl_output";
+			} else if (url != NULL &&
+				   strncmp(url, RTMP_PROTOCOL,
+					   strlen(RTMP_PROTOCOL)) != 0) {
+				type = "ffmpeg_mpegts_muxer";
+			}
```

Comment:
I think primary and backup server share the same type of output, the type has no need to be set multiple times.